src/nvim/ops.c

```diff
@@ -3644,6 +3644,10 @@ int do_join(size_t count,
    */
   for (t = (linenr_T)count - 1;; t--) {
     cend -= currsize;
+	if(cend == NULL || curr == NULL){
```

Comment:
There's no need to merge to your local master branch.  Work is typically done on your computer in a distinct feature branch.  That feature branch is pushed to Github.  When/if the project maintainers decide to include the change, they will merge it into master (or whatever the target branch is).